pkg/core/blockchain.go

```diff
@@ -285,14 +285,14 @@ func NewBlockchain(s storage.Store, cfg config.Blockchain, log *zap.Logger) (*Bl
 	}
 	if cfg.Hardforks == nil {
 		cfg.Hardforks = map[string]uint32{}
-		for _, hf := range config.Hardforks {
+		for _, hf := range config.StableHardforks {
 			cfg.Hardforks[hf.String()] = 0
 		}
-		log.Info("Hardforks are not set, using default value")
+		log.Info("Hardforks are not set, enabling all stable ones", zap.Stringer("latest", config.HFLatestStable))
 	} else if len(cfg.Hardforks) != 0 {
 		// Explicitly set the height of all old omitted hardforks to 0 for proper
 		// IsHardforkEnabled behaviour.
-		for _, hf := range config.Hardforks {
+		for _, hf := range config.StableHardforks {
 			if _, ok := cfg.Hardforks[hf.String()]; !ok {
```

Comment:
I don't think that this change is appropriate. Consider case when we have two unstable hardforks, second of which is included into the node's config. Then this logic won't include first in the node config, although the list of enabled hardforks must be continious.

It's a nitpicking, hardly ever we're going to have two unstable hardforks, but we'd better fix the code.